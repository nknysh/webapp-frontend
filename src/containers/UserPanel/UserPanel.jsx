import React, { useState } from 'react';
import { compose, path, map, lensPath, set, values } from 'ramda';

import uiConfig from 'config/ui';
import userPanelLinks from 'config/links/user-panel';

import { DropDownMenu, Link } from 'components';
import { withAuthentication } from 'hoc';

import connect from './UserPanel.state';
import { propTypes, defaultProps } from './UserPanel.props';
import { StyledUserPanel, Text } from './UserPanel.styles';
import { isSending } from 'store/common';

const logoutClickLens = lensPath(['logout', 'onClick']);

// eslint-disable-next-line
const renderLink = ({ title, ...props }) => (
  <Link key={title} {...props}>
    {title}
  </Link>
);

export const UserPanel = ({ isAuthenticated, currentUser, requestStatus, token, onLogout }) => {
  if (!isAuthenticated || !currentUser) return null;

  const [logout, setLogout] = useState(false);

  const { firstName, lastName } = currentUser;

  const onLogoutClick = () => {
    setLogout(true);
    onLogout(token);
  };

  const links = set(logoutClickLens, onLogoutClick, userPanelLinks);

  if (logout && isSending(requestStatus))
    return (
      <DropDownMenu
        showArrow={false}
        title={<Text data-placeholder>{path(['messages', 'loggingOut'], uiConfig)}</Text>}
      />
    );

  return (
    <StyledUserPanel>
      <DropDownMenu
        title={
          <Text>
            {firstName} {lastName}
          </Text>
        }
      >
        {values(map(renderLink, links))}
      </DropDownMenu>
    </StyledUserPanel>
  );
};

UserPanel.propTypes = propTypes;
UserPanel.defaultProps = defaultProps;

export default compose(
  withAuthentication,
  connect
)(UserPanel);
