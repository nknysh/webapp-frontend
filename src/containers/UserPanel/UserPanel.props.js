import PropTypes from 'prop-types';

export const propTypes = {
  isAuthenticated: PropTypes.bool,
};

export const defaultProps = {};
