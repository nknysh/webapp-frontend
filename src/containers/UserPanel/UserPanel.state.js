import { connect } from 'react-redux';
import { getAuthStatus, getAuthError, getCurrentUser } from 'store/modules/auth/selectors';

export const mapStateToProps = state => ({
  requestStatus: getAuthStatus(state),
  error: getAuthError(state),
  currentUser: getCurrentUser(state),
});

export default connect(mapStateToProps);
