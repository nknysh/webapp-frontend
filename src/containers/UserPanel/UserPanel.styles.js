import styled, { css, keyframes } from 'styled-components';
import { Select, Link as BaseLink } from '@pure-escapes/webapp-ui-components';

import { theme } from 'styles';
import { pureUiTheme } from 'pureUi/pureUiTheme';

const dropdownIn = keyframes`
  from {
    opacity: 0;
    top: 15px;
  }

  to {
    opacity: 1;
    top: 30px;
  }
`;

export const StyledUserPanel = styled.div`
  position: relative;
  padding: ${theme.spacing.gutter}px;

  ${props => props.theme.breakpoints.tablet`
    padding: 0;
  `}

  .link {
    color: ${pureUiTheme.colors.black};
    font-size: 16px;
    display: block;
  }

  ul.dropdown {
    animation: ${dropdownIn} 0.25s ease-out;
    border-radius: 8px;
    position: absolute;
    right: 0;
    top: 30px;
    background-color: white;
    padding: 10px 0;
    list-style: none;
    margin: 0;
    box-shadow: 0 0 5px rgba(0, 0, 0, 0.1);
    border: ${pureUiTheme.colorRoles.lightGreyBorder} 1px solid;
    color: ${pureUiTheme.colors.black};
    font: ${pureUiTheme.typography.sansSerifFont};
    text-transform: uppercase;
  }

  ul.dropdown > li {
    font-size: 16px;
    padding: 10px;
    border-bottom: ${pureUiTheme.colorRoles.lightGreyBorder} 1px solid;

    &:last-child {
      border: none;
    }
  }

  .dropdownBG {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-color: rgba(0, 0, 0, 0);
  }
`;

export const Text = styled.div`
  margin: ${theme.spacing.gutter / 2}px 0;
  padding: ${theme.spacing.gutter / 2}px 0;
  text-transform: uppercase;
  font-weight: ${theme.fonts.bold};
  cursor: pointer;
  transition: ${theme.defaultTransition};

  &:hover {
    color: ${theme.palette.primary};
  }

  ${({ ['data-placeholder']: placeholder }) =>
    placeholder &&
    css`
      color: ${theme.colors.gray};
    `}

  ${props => props.theme.breakpoints.tablet`
      margin-left: ${theme.spacing.gutter}px;
      padding-left: ${theme.spacing.gutter}px;
    `}
`;

export const Country = styled.div``;
export const CountrySelect = styled(Select)``;

export const Link = styled(BaseLink)`
  width: 100%;
  flex: 0;

  ${props => props.theme.breakpoints.desktop`
      flex: 1;
    `}
`;
