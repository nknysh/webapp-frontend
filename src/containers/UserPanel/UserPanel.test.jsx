import React from 'react';

import { UserPanel } from './UserPanel';

const getComponent = props => shallow(<UserPanel isLoading={false} {...props} />);

describe('<UserPanel />', () => {
  describe('render', () => {
    it('matches snapshot', () => {
      expect(getComponent()).toMatchSnapshot();
    });
  });
});
