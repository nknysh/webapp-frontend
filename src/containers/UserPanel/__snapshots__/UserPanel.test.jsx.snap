// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserPanel /> render matches snapshot 1`] = `""`;
